src/core/Akka.Tests/Actor/Bugfix7501Specs.cs

```diff
@@ -0,0 +1,64 @@
+// -----------------------------------------------------------------------
+//  <copyright file="Bugfix7501Specs.cs" company="Akka.NET Project">
+//      Copyright (C) 2009-2025 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2025 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Actor.Dsl;
+using Akka.TestKit;
+using Akka.TestKit.TestActors;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Tests.Actor;
+
+public class Bugfix7501Specs : AkkaSpec
+{
+    public Bugfix7501Specs(ITestOutputHelper output) : base(output)
+    {
+        
+    }
+
+    [Fact]
+    public async Task FutureActorRefShouldSupportDeathWatch()
```

Comment:
Handles both scenarios: 

1. `Context.Watch` before the `FutureActorRef<T>` completes
2. `Context.Watch` after the `FutureActorRef<T>` has already completed, which should immediately report back with a `Terminated` message